### PR TITLE
Build custom libtar in conda

### PR DESCRIPTION
**Pull request by JanuszL**
- build custom libtar to include the latest patches on top of
  1.2.20, including one fixing reading of empty archives

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

## Description
- [x] **Bug fix** (*non-breaking change which fixes an issue*)
- [ ] **New feature** (*non-breaking change which adds functionality*)
- [ ] **Breaking change** (*fix or feature that would cause existing functionality to not work as expected*)
- [ ] **Refactoring** (*Redesign of existing code that doesn't affect functionality*)
- [ ] **Other** (*e.g. Documentation, Tests, Configuration*)

### What happened in this PR
- build custom libtar to include the latest patches on top of 1.2.20, including one fixing reading of empty archives

#### Additional information
- Affected modules and functionalities:
  - conda build

- Key points relevant for the review:
  - NA

### Tests
- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: conda/third_party/dali_libtar/recipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{% set name = "dali-libtar" %}
```
Comment (klecki) [inline conda/third_party/dali_libtar/recipe/meta.yaml:1]: Add license? We have it in yml for ffmpeg

Comment (klecki) [reply conda/third_party/dali_libtar/recipe/meta.yaml:1]: Or is this copied from conda?

Comment (JanuszL) [reply conda/third_party/dali_libtar/recipe/meta.yaml:1]: Copied from https://github.com/conda-forge/libtar-feedstock.

File: conda/recipe/meta.yaml [hunk 2]
```diff
@@ -93,7 +93,6 @@ requirements:
     - lmdb >=0.9.28
     - libtiff >=4.3.0
     - libsndfile >=1.0.31
-    - libtar >=1.2.20
```
Comment (JanuszL) [inline conda/recipe/meta.yaml:96]: libtar provides a static variant and we link with it so we don't need that in runtime.

File: conda/third_party/dali_libtar/recipe/meta.yaml (hunk 1 above)
Comment (klecki) [reply conda/third_party/dali_libtar/recipe/meta.yaml:1]: How about their license? Do we need any Acknowledgments?

Comment (JanuszL) [reply conda/third_party/dali_libtar/recipe/meta.yaml:16]: Added

**Review by klecki — APPROVED** [review]

**Review by awolant — APPROVED** [review]

**Pull request merged**